### ISSUE TITLE
Regression in latest Test Resources

### ISSUE THREAD

**Issue by sdelamo** [micronaut-projects/micronaut-test-resources#243]
### Issue description

I updated [platform to test-resources M3](https://github.com/micronaut-projects/micronaut-platform/pull/276)

I discovered a regression while running starter tests which involve test resources. 

## Steps to reproduce 

[Git clone 4.0.x branch in micronaut-starter](https://github.com/micronaut-projects/micronaut-starter/tree/4.0.x)

Generate an application with `mysql` and `data-hibernate-reactive` features which adds Test Resources by default. 

```
./gradlew micronaut-cli:run --args="create-app --jdk 17 -b gradle -f \"mysql,data-hibernate-reactive\" temp"
```

Run tests: 

```
cd starter-cli/temp
./gradlew test
```

You will get: 

```

> Task :compileJava
> Task :processResources
> Task :classes
> Task :compileTestJava
Note: Creating bean classes for 1 type elements
> Task :inspectRuntimeClasspath
> Task :compileTestResourcesJava NO-SOURCE
> Task :processTestResourcesResources NO-SOURCE
> Task :testResourcesClasses UP-TO-DATE
[34m[test-resources-service][0;39m [36m09:48:49.303[0;39m [1;30m[ForkJoinPool.commonPool-worker-2][0;39m [34mINFO [0;39m [35mi.m.t.e.TestResourcesResolverLoader[0;39m - Loaded 3 test resources resolvers: io.micronaut.testresources.mysql.MySQLTestResourceProvider, io.micronaut.testresources.hibernate.reactive.mysql.HibernateReactiveMySQLTestResourceProvider, io.micronaut.testresources.testcontainers.GenericTestContainerProvider
[34m[test-resources-service][0;39m [36m09:48:49.319[0;39m [1;30m[main][0;39m [1;31mERROR[0;39m [35mio.micronaut.runtime.Micronaut[0;39m - Error starting Micronaut server: Bean definition [io.micronaut.http.server.netty.binders.NettyBinderRegistrar] could not be loaded: 'void io.micronaut.context.AbstractInitializableBeanDefinition$PrecalculatedInfo.<init>(java.util.Optional, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean)'
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [io.micronaut.http.server.netty.binders.NettyBinderRegistrar] could not be loaded: 'void io.micronaut.context.AbstractInitializableBeanDefinition$PrecalculatedInfo.<init>(java.util.Optional, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean)'
	at io.micronaut.context.DefaultBeanContext$BeanDefinitionProducer.getDefinition(DefaultBeanContext.java:4213)
	at io.micronaut.context.DefaultBeanContext.collectBeanCandidates(DefaultBeanContext.java:2198)
	at io.micronaut.context.DefaultBeanContext.findBeanCandidates(DefaultBeanContext.java:2173)
	at io.micronaut.context.DefaultBeanContext.findBeanCandidatesInternal(DefaultBeanContext.java:3432)
	at io.micronaut.context.DefaultBeanContext.getBeanDefinitions(DefaultBeanContext.java:838)
	at io.micronaut.context.DefaultBeanContext.getBeanDefinitions(DefaultBeanContext.java:832)
	at io.micronaut.context.DefaultBeanContext.getTypeToListenerMap(DefaultBeanContext.java:1990)
	at io.micronaut.context.DefaultBeanContext.loadListeners(DefaultBeanContext.java:1949)
	at io.micronaut.context.DefaultBeanContext.initializeEventListeners(DefaultBeanContext.java:1935)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3409)
	at io.micronaut.context.DefaultBeanContext.finalizeConfiguration(DefaultBeanContext.java:3764)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:345)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:191)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:75)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:324)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:310)
	at io.micronaut.testresources.server.TestResourcesService.main(TestResourcesService.java:41)
Caused by: java.lang.NoSuchMethodError: 'void io.micronaut.context.AbstractInitializableBeanDefinition$PrecalculatedInfo.<init>(java.util.Optional, boolean, boolean, boolean, boolean, boolean, boolean, boolean, boolean)'
	at io.micronaut.http.server.netty.binders.$NettyBinderRegistrar$Definition.<init>(Unknown Source)
	at io.micronaut.http.server.netty.binders.$NettyBinderRegistrar$Definition.<init>(Unknown Source)
	at io.micronaut.http.server.netty.binders.$NettyBinderRegistrar$Definition$Reference.load(Unknown Source)
	at io.micronaut.context.AbstractInitializableBeanDefinitionReference.load(AbstractInitializableBeanDefinitionReference.java:178)
	at io.micronaut.context.DefaultBeanContext$BeanDefinitionProducer.getDefinition(DefaultBeanContext.java:4208)
	... 16 common frames omitted
> Task :internalStartTestResourcesService
``

**Comment by sdelamo** [comment]
@melix do you know about this?

**Comment by melix** [comment]
I cannot even build the command to generate the test project: https://ge.micronaut.io/s/kdown3pj2xr3w/console-log?anchor=260&page=1

**Comment by sdelamo** [comment]
Is this with branch 4.0.x ?

**Comment by melix** [comment]
Ok so the issue is just that the Gradle plugin defines the default `testResourcesService` to 2.0.0-M1, which is incompatible. You can workaround with:
```
micronaut {
    runtime("netty")
    testRuntime("junit5")
    processing {
        incremental(true)
        annotations("temp.*")
    }
    testResources {
        version = "2.0.0-M3"
        additionalModules.add("hibernate-reactive-mysql")
    }
}
```

Or we can release a new milestone of the Gradle plugin with the upgrade.

**Comment by sdelamo** [comment]
I think we should release a milestone of the Gradle plugin. @melix can you do that?

**Comment by melix** [comment]
Yes

**Comment by sdelamo** [comment]
I have verified this is solved with 4.0.0-M2.
